CMakeLists.txt

```diff
@@ -181,9 +181,6 @@ endif()
 
 if(MSVC)
   target_compile_options(onnx_proto PRIVATE /WX-)
-  if(BUILD_SHARED_LIBS)
```

Comment:
We don't need to define `ONNX_API` manually as it's handled here: https://github.com/onnx/onnx/blob/b184dd3cb80f9f0ab47719ed261d46cdec0f697b/onnx/onnx_pb.h#L23-L42

The correct way is to set `ONNX_BUILD_SHARED_LIBS` and `ONNX_BUILD_MAIN_LIB` during CMake.